pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1244,32 +1244,30 @@ func GetPodAntiAffinityTerms(podAntiAffinity *v1.PodAntiAffinity) (terms []v1.Po
 	return terms
 }
 
-// getMatchingAffinityTerms returns the given affinityKind terms of the pod that matches the target pod labels
-func getMatchingAffinityTerms(pod, targetPod *v1.Pod, affinityKind string) (matchedTerms []v1.PodAffinityTerm, err error) {
+// getMatchingAffinityTerms returns the given "affinityProp" terms of the pod that matches the target pod labels, and the pod`s affinity number for further use.
+func getMatchingAffinityTerms(pod, targetPod *v1.Pod, affinityProp bool) (matchedTerms []v1.PodAffinityTerm, podAffinityNum int, err error) {
```

Comment:
I don't think `podAffinityNum` is necessary. The caller can get the number in O(1) via `len(matchedTerms)`, or do I missing anything?

Reply:
podAffinityNum is the number of affinity defined in podSpec, the number of matchedTerm may be less or equal with the number of podAffinity, So in the matched all sementic , we need podAffinityNum to help us verify